sourcegraph/learn
# Update dependencies

### What should this PR do?
Resolves [DEVED-93](https://sourcegraph.atlassian.net/browse/DEVED-93) by bumping versions of `js-yaml`, `next-mdx-remote`, `rehype-react`, `remark-rehype`, and `sass`.

This PR does *not* bump `rehype-slug` or `rehype-autolink-headings` due to low current adoption of these packages currently (in each case, a Sourcegraph search results in only one instance of adoption for these versions in 1M repos), coupled with the update being a breaking change. 

Some context for future discussion of adopting these packages:
[`rehype-slug` rewrite](https://github.com/rehypejs/rehype-slug/commit/b293ef3e48d2aa87f0ccb3f5d34aa3c096e0ba80)
[`next-mdx-remote` type definitions, particularly for `mdxOptions`](https://github.com/hashicorp/next-mdx-remote/blob/452755393f2bcadf1807f6398bf1429069acba42/src/types.ts#L13)
[Open issue](https://github.com/hashicorp/next-mdx-remote/issues/86) on types for `remarkPlugins` and `rehypePlugins` in `next-mdx-remote`.

### Why are we making this change?
- Part of an effort to keep our dependencies up-to-date

### What are the acceptance criteria? 
- The app should exhibit no changes, and builds should succeed.

### How should this PR be tested?
- Check out the branch, remove `node_modules`, and install dependencies with `npm ci`.
- Ensure that the above criteria are met.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
